group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTestContext.java

```diff
@@ -0,0 +1,1485 @@
+        int classicGroupInitialRebalanceDelayMs,
+        int classicGroupNewMemberJoinTimeoutMs
+    ) {
+        this.time = time;
+        this.timer = timer;
+        this.snapshotRegistry = snapshotRegistry;
+        this.metrics = metrics;
+        this.groupMetadataManager = groupMetadataManager;
+        this.classicGroupInitialRebalanceDelayMs = classicGroupInitialRebalanceDelayMs;
+        this.classicGroupNewMemberJoinTimeoutMs = classicGroupNewMemberJoinTimeoutMs;
+        snapshotRegistry.getOrCreateSnapshot(lastWrittenOffset);
+    }
+
+    public void commit() {
+        long lastCommittedOffset = this.lastCommittedOffset;
+        this.lastCommittedOffset = lastWrittenOffset;
+        snapshotRegistry.deleteSnapshotsUpTo(lastCommittedOffset);
+    }
+
+    public void rollback() {
+        lastWrittenOffset = lastCommittedOffset;
+        snapshotRegistry.revertToSnapshot(lastCommittedOffset);
+    }
+
+    public ConsumerGroup.ConsumerGroupState consumerGroupState(
+        String groupId
+    ) {
+        return groupMetadataManager
+            .getOrMaybeCreateConsumerGroup(groupId, false)
+            .state();
+    }
+
+    public ConsumerGroupMember.MemberState consumerGroupMemberState(
+        String groupId,
+        String memberId
+    ) {
+        return groupMetadataManager
+            .getOrMaybeCreateConsumerGroup(groupId, false)
+            .getOrMaybeCreateMember(memberId, false)
+            .state();
+    }
+
+    public CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGroupHeartbeat(
+        ConsumerGroupHeartbeatRequestData request
+    ) {
+        RequestContext context = new RequestContext(
+            new RequestHeader(
+                ApiKeys.CONSUMER_GROUP_HEARTBEAT,
+                ApiKeys.CONSUMER_GROUP_HEARTBEAT.latestVersion(),
+                "client",
+                0
+            ),
+            "1",
+            InetAddress.getLoopbackAddress(),
+            KafkaPrincipal.ANONYMOUS,
+            ListenerName.forSecurityProtocol(SecurityProtocol.PLAINTEXT),
+            SecurityProtocol.PLAINTEXT,
+            ClientInformation.EMPTY,
+            false
+        );
+
+        CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> result = groupMetadataManager.consumerGroupHeartbeat(
+            context,
+            request
+        );
+
+        result.records().forEach(this::replay);
+        return result;
+    }
+
+    public List<MockCoordinatorTimer.ExpiredTimeout<Void, Record>> sleep(long ms) {
+        time.sleep(ms);
+        List<MockCoordinatorTimer.ExpiredTimeout<Void, Record>> timeouts = timer.poll();
+        timeouts.forEach(timeout -> {
+            if (timeout.result.replayRecords()) {
+                timeout.result.records().forEach(this::replay);
+            }
+        });
+        return timeouts;
+    }
+
+    public MockCoordinatorTimer.ScheduledTimeout<Void, Record> assertSessionTimeout(
```

Comment:
The return value here doesn't seem to be used by any of the tests can't we just return void?